fix(web-analytics): connect data logics to scene logic

mariusandra opened the pull request:
## Problem

The web analytics tab would reload every time it was opened.

## Changes

Attach the various `dataNodeLogic`s to the scene's logic.

## How did you test this code?

Clicked between the tabs, saw the count of queries to `/query` go from a dozen to just one. Still a bit of work to do to remove that last one.

mariusandra committed:
fix(web-analytics): connect data logics to scene logic

adamleithp reviewed (approved):
ts error, but :check:

mariusandra committed:
tsc

mariusandra committed:
more

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

rafaeelaudibert reviewed (approved):
I mean, it works, but I don't really understand why. I'd love a brownbag (or just a short video) going over the reasons why this works/is needed

I'll also do a follow-up PR simplifying that `LogicWrapper | BuiltLogic` type

mariusandra committed:
Merge branch 'master' into web-dash-tabs

mariusandra committed:
Merge branch 'master' into web-dash-tabs

mariusandra committed:
Merge branch 'master' into web-dash-tabs

rafaeelaudibert committed:
Merge branch 'master' into web-dash-tabs

rafaeelaudibert committed:
refactor: Add `SceneLogicAttachTo` helper

This replaces `LogicWrapper | BuiltLogic`

rafaeelaudibert committed:
fix: Connect to the `preflightLogic`

I don't really understand why this is needed, but Storybook *needs* this for it to work. This happens because we're building `revenueAnalyticsLogic` on RevenueAnalyticsScene.stories.tsx above the `App`. Another possible fix is to simply load `preflightLogic` in that file too, but this one feels slightly cleaner.

rafaeelaudibert commented:
@mariusandra I merged on master and added a new commit simplifying that whole `LogicWrapper | BuiltLogic` thing with a simpler type.

I've also fixed the [Storybook failure](https://github.com/PostHog/posthog/actions/runs/17781838402/job/50542215281?pr=38008). I don't really understand  **why** it failed or why [18ef289](https://github.com/PostHog/posthog/pull/38008/commits/18ef28900c9331e24bd9920edf163cb35157a803) fixes it. The error happens because our story instantiates `revenueAnalyticsLogic` (it stops happening if we remove that). Any idea why we need that mounted? I don't really see the reason

I also tried forcing the `preflightLogic`  to be always mounted, but that didn't solve it. That bug does not happen in the app.

mariusandra commented on frontend/src/lib/logic/scenes/useAttachedLogic.ts:
```diff
@@ -1,12 +1,15 @@
 import { useEffect, useState } from 'react'
 
+// Type used to attach a logic to the scene logic
+export type SceneLogicAttachTo<LogicType extends Logic = Logic> = BuiltLogic<LogicType> | LogicWrapper<LogicType>
```
I'd rather we not do this. `LogicWrapper` and `BuiltLogic` are the two shapes a logic can take in Kea (before building and after). I personally don't understand why must we abstract away them away... And if we really want to, this has nothing to do with `AttachTo` --> I'd just call it `AnyKindOfLogic` or something like that.

mariusandra replied:
I also replaced the "fix" to preflightLogic. Your scene's logic has nothing to do with preflightLogic, so I don't see how is linking them for no reason a cleaner solution. We should fix whatever is causing storybook to setup a weird environment where the logics don't work.

mariusandra committed:
Revert "refactor: Add `SceneLogicAttachTo` helper"

This reverts commit 0729fd078fdee07f54e3b0736a31d4c472c5998a.

mariusandra committed:
Revert "fix: Connect to the `preflightLogic`"

This reverts commit 18ef28900c9331e24bd9920edf163cb35157a803.

mariusandra committed:
Merge remote-tracking branch 'origin' into web-dash-tabs

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

mariusandra committed:
connect preflightLogic to appLogic

mariusandra committed:
Merge remote-tracking branch 'origin/web-dash-tabs' into web-dash-tabs
